src/tokens/selectors.ts

```diff
@@ -50,6 +50,22 @@ export const tokensWithUsdValueSelector = createSelector(tokensListSelector, (to
   ) as TokenBalanceWithUsdPrice[]
 })
 
+export const stalePriceSelector = createSelector(tokensListSelector, (tokens) => {
+  // If tokens with usd value are present check the time price was fetched
+  // Else usd values are not present so we know prices are invalid
+  const tokensWithUsdValue = tokens.filter((tokenInfo) => tokenInfo.usdPrice !== null)
+  if (tokensWithUsdValue.length !== 0) {
+    return (
+      tokensWithUsdValue.filter(
+        (tokenInfo) =>
+          (tokenInfo.priceFetchedAt ?? 0) < Date.now() - TIME_UNTIL_TOKEN_INFO_BECOMES_STALE
+      ).length > 0
+    )
+  } else {
+    return true
+  }
```

Comment:
it took my brain a little time to understand this, do you think this alternative is the same/easier to read?
```
return tokens.every(tokenInfo => tokenInfo.usdPrice === null) || tokens.some(tokenInfo => (tokenInfo.priceFetchedAt ?? 0) < Date.now() - TIME_UNTIL_TOKEN_INFO_BECOMES_STALE)
```

Reply:
Tried the exact code and prices where showing as stale when not expected. Will take another look, but this looks like it needs a bit of tweaking.

Reply:
Added some clarifying comments in https://github.com/valora-inc/wallet/pull/2645/commits/86fd596d2b008ea288201660048c4385bab05966!